src/runtime_config.rs

```diff
@@ -0,0 +1,94 @@
+use serde::Deserialize;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::fs;
+use tokio::sync::RwLock;
+use tokio::task::JoinHandle;
+use tracing::{error, info};
+
+#[derive(Debug, Deserialize, Clone, PartialEq, Default)]
+pub struct RuntimeConfig {
+    /// A list of tasks to drop before inserting into sqlite.
+    pub drop_task_killswitch: Vec<String>,
+}
+
+pub struct RuntimeConfigManager {
+    pub config: Arc<RwLock<RuntimeConfig>>,
+    pub handler: JoinHandle<()>,
+}
+
+impl RuntimeConfigManager {
+    pub async fn new(path: Option<String>) -> Self {
+        let runtime_config = Arc::new(RwLock::new(Default::default()));
+        let _ = Self::reload_config(&path, &runtime_config).await;
+        Self {
+            config: runtime_config.clone(),
+            handler: tokio::spawn(async move {
+                let mut interval = tokio::time::interval(Duration::from_secs(60));
+                loop {
+                    Self::reload_config(&path, &runtime_config).await;
+                    interval.tick().await;
+                }
+            }),
+        }
+    }
+
+    async fn reload_config(path: &Option<String>, config: &Arc<RwLock<RuntimeConfig>>) {
+        if path.is_none() {
+            return;
+        }
+        let yaml_path = path.as_ref().unwrap();
+        let contents = fs::read_to_string(yaml_path).await;
+        let new_config: RuntimeConfig = match contents {
```

Comment:
I think there is a case here that needs to be addressed:

- We successfully set some killswitches
- We then want to add something else to the list, but mess up the yaml

In that case, all the original changes will be overridden by the default config. In that case I would rather the server continue using the existing config, instead of the default.

Reply:
That's a good point. Updated to use existing runtime configs instead of overriding with defaults if yaml is malformed. 👍